### PR TITLE
Align gradle with maven plugin

### PR DESCRIPTION
**Pull request by wilmveel**

### CODE REVIEW

**Review by sjorsdev — APPROVED** [review]

File: src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt [hunk 1]
```diff
@@ -35,14 +34,21 @@ class WirespecPlugin : Plugin<Project> {
...
 
-    private fun compile(input: String, logger: Logger, emitter: Emitter) =
-        (File(input).listFiles() ?: arrayOf())
-            .map { it.name.split(".").first() to it.bufferedReader(Charsets.UTF_8) }
-            .map { (name, reader) -> name to WirespecSpec.compile(reader.collectToString())(logger)(emitter) }
+    private fun String.readFiles() = (File(this).listFiles() ?: arrayOf())
+    private fun Array<File>.parse() = this
+        .map { it.name.split(".").first() to it.bufferedReader(Charsets.UTF_8) }
+        .map { (name, reader) -> name to WirespecSpec.parse(reader.collectToString())(logger) }
+    private fun List<Emitted>.writeFiles() = this
```
Comment (jerrevanveluw) [inline src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt:41]: Why return `this`?

File: src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt [hunk 2]
```diff
@@ -68,20 +80,64 @@ class WirespecPlugin : Plugin<Project> {
...
+                        JavaShared
+                    )
+                }
+                if (Language.Scala in languages) {
+                    ScalaEmitter(packageName, logger).emit(
+                        input,
+                        output,
+                        Scala,
+                        PackageName(packageName),
+                        KotlinShared
```
Comment (jerrevanveluw) [inline src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt:119]: Should be ScalaShared

File: src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt [hunk 3]
```diff
@@ -68,20 +80,64 @@ class WirespecPlugin : Plugin<Project> {
...
+                    )
+                }
+                if (Language.TypeScript in languages) {
+                    TypeScriptEmitter(logger).emit(
+                        input,
+                        output,
+                        TypeScript,
+                        PackageName(packageName),
+                        KotlinShared
```
Comment (jerrevanveluw) [inline src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt:128]: Should be null or TypeScriptShared

File: src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt [hunk 4]
```diff
@@ -68,20 +80,64 @@ class WirespecPlugin : Plugin<Project> {
...
+                    )
+                }
+                if (Language.Wirespec in languages) {
+                    WirespecEmitter(logger)
+                        .emit(
+                            input,
+                            output,
+                            Wirespec,
+                            PackageName(packageName),
+                            KotlinShared
```
Comment (jerrevanveluw) [inline src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt:138]: Should be null or WirespecShared

**Review by jerrevanveluw — CHANGES_REQUESTED** [review]
If the only thing you are adding is the custom emitter, we should include that in the main ktor example. Then that example has just one more emitter...

File: src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt (hunk 1 above)
Comment (wilmveel) [reply src/plugin/gradle/src/main/kotlin/WirespecPlugin.kt:42]: Function was never used

**Review by jerrevanveluw — APPROVED** [review]

**Pull request merged**